### PR TITLE
refactor: add product.service.js, product.repository.js

### PR DESCRIPTION
**Pull request by maximsan**
**Evaluation criteria**

- [x] - File serverless.yml contains configuration for catalogBatchProcess function
- [x] - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [x] - File serverless.yml contains configuration for SQS catalogItemsQueue
- [x] - File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription

**Additional (optional) tasks**
- [x] - catalogBatchProcess lambda is covered by unit tests
- [x] - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

FE  - [link](https://dg2e8nno6k0ty.cloudfront.net/)

### CODE REVIEW

File: product-service/src/handlers/catalogBatchProcess.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    try {
+        await ProductServ.create(products);
+    } catch (error) {
+        console.log('Error during product creation');
+        return errorHandler(error);
+    }
+
+    try {
+        await SnsServ.send(products);
```
Comment (BaturevichNikita) [inline product-service/src/handlers/catalogBatchProcess.js]: Looks like this lambda sends a message on each request, isn't it? 
But in clause 6.3 it is written that the letter must be sent once.

Comment (maximsan) [reply product-service/src/handlers/catalogBatchProcess.js]: I'm not sure about your note. I guess we send message after every group of created products.

Comment (BaturevichNikita) [reply product-service/src/handlers/catalogBatchProcess.js]: image
Seems, that we should use a FIFO queue

Comment (BaturevichNikita) [reply product-service/src/handlers/catalogBatchProcess.js]: This is just my opinion, the evaluation criteria says that the function should send letters, so I'm not going to underestimate the score =)

Comment (maximsan) [reply product-service/src/handlers/catalogBatchProcess.js]: Thanks that highlight it. Lector mentioned that we use only **standard** queue in the task. It worth to mention that this is notification service, the queue is used in other one. By the way we can continue our chatting via **discord**.

**Pull request merged**